cmd/spiderpool/cmd/cmd_suite_test.go

```diff
@@ -26,25 +31,86 @@ func TestCmd(t *testing.T) {
 }
 
 var _ = BeforeSuite(func() {
+	// restore the ENV
+	DeferCleanup(os.Setenv, socketPathEnvName, os.Getenv(socketPathEnvName))
+
+	t := GinkgoT()
+	tempDir := t.TempDir()
+
+	// In concurrency situation, we need use different unix socket path.
+	err = os.Setenv(socketPathEnvName, tempDir+"/tmp.sock")
+	Expect(err).NotTo(HaveOccurred())
+
+	// refresh singleton agentContext unix socket path.
+	cmd.GetAgentContext().RegisterEnv()
+
 	unixServer, err = cmd.NewAgentOpenAPIUnixServer()
 	Expect(err).NotTo(HaveOccurred())
 
+	err := os.RemoveAll(string(unixServer.SocketPath))
+	Expect(err).NotTo(HaveOccurred())
+
+	startWg := sync.WaitGroup{}
+	startWg.Add(1)
+
+	// start unix server.
 	go func() {
+		defer GinkgoRecover()
+
+		startWg.Done()
 		if err = unixServer.Serve(); nil != err {
 			if err == net.ErrClosed {
 				return
 			}
-			By(err.Error())
-			return
+			By("Error: spider agent server finished with error: " + err.Error())
 		}
 	}()
 
-	// [must] give client some time to wait for the unix server starts completely
-	// or the client will request fail cause the server doesn't serve successfully.
-	time.Sleep(time.Second)
+	startWg.Wait()
+
+	if !checkServer(string(unixServer.SocketPath)) {
```

Comment:
transport := &http.Transport{
 		DisableCompression: true,
 		Dial: func(_, _ string) (net.Conn, error) {
 			return net.Dial("unix", sockPath)
 		},
 	}
 httpClient := &http.Client{Transport: transport}

Eventually(func(g Gomega) bool {
				defer GinkgoRecover()
				httpClient.Get("http://localhost/")
                                 .......
				return true
			}).WithTimeout(10 * time.Second).WithPolling(1 * time.Second).Should(BeTrue())

Reply:
>  However, if your spec launches a goroutine that calls Fail (or, equivalently, invokes a failing Gomega assertion), there's no way for Ginkgo to rescue the panic that Fail throws. This will cause the suite to panic and no subsequent specs will run. To get around this you must rescue the panic using defer GinkgoRecover()

此处无需使用defer GinkgoRecover()